src/Core/src/Handlers/Application/ApplicationHandler.Android.cs

```diff
@@ -1,15 +1,23 @@
+using System;
 using Android.App;
-using Microsoft.Extensions.Logging;
-using Microsoft.Maui.LifecycleEvents;
-using Microsoft.Maui.Platform;
 
 namespace Microsoft.Maui.Handlers
 {
 	public partial class ApplicationHandler : ElementHandler<IApplication, Application>
 	{
 		public static void MapTerminate(ApplicationHandler handler, IApplication application, object? args)
 		{
-			handler.Logger?.LogWarning("Android does not support programmatically terminating the app.");
+			var currentActivity = ApplicationModel.Platform.CurrentActivity;
+
+			if (currentActivity != null)
+			{
+				if (OperatingSystem.IsAndroidVersionAtLeast(21))
```

Comment:
MAUI only supports 21+ already, so I don't think we need an extra check here.... Even after reading the android documentation, I'm still not super clear on which finish variant is the 'best' one here, but presumably if this code came from Forms, maybe we can track down the conversation/comments/PR around choosing it to be sure?

Reply:
Updated. Have removed the version validation.